src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/segmentgroup/SG01.java

```diff
@@ -0,0 +1,76 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.PerformingOrganisationNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RequesterNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ServiceProvider;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class SG01 extends SegmentGroup {
```

Comment:
So these were all just called `SGNN` (with `NN` being the number), but probably we can give decent semantic names reflecting the purpose/contents of each segment group. E.g., the document says the segment group 1 is
> To represent information about involved healthcare parties

So perhaps this could be
```suggestion
public class InvolvedParty extends SegmentGroup {
```